app/src/main/kotlin/pl/elpassion/report/Report.kt

```diff
@@ -1,13 +1,38 @@
 package pl.elpassion.report
 
+import pl.elpassion.project.Project
 import java.io.Serializable
 
-data class Report(
-        val id : Long,
-        val year: Int,
-        val month: Int,
-        val day: Int,
+data class HourlyReport(
+        override val id: Long,
+        override val year: Int,
+        override val month: Int,
+        override val day: Int,
         val reportedHours: Double,
-        val projectName: String,
-        val projectId: Long,
-        val description: String) : Serializable
\ No newline at end of file
+        val project: Project?,
+        val description: String,
+        val reportType: HourlyReportType) : Serializable, Report
+
+data class DailyReport(
+        override val id: Long,
+        override val year: Int,
+        override val month: Int,
+        override val day: Int,
+        val reportType: DailyReportType) : Serializable, Report
+
+interface Report {
+    val id: Long
+    val year: Int
+    val month: Int
+    val day: Int
+}
+
+enum class HourlyReportType {
+    REGULAR,
+    PAID_VACATIONS,
+}
+
+enum class DailyReportType {
+    UNPAID_VACATIONS,
+    SICK_LEAVE
+}
```

Comment:
I am still thinking about this model and it is evolving in my head. It's not trivial thing to do it right. Btw it reminds me one of the Clean Code episode.